ArchiSteamFarm/Bot.cs

```diff
@@ -309,6 +311,27 @@ internal sealed class Bot : IDisposable {
 			return await BotDatabase.MobileAuthenticator.HandleConfirmations(confirmations, accept).ConfigureAwait(false);
 		}
 
+		private async Task AcceptDigitalGiftCards() {
```

Comment:
This function can be called multiple times due to multiple events generated by Steam network. You should implement appropriate semaphore + queue, check `ArchiWebHandler.MarkInventory()` or `Trading.OnNewTrade()` for examples how to achieve that.

Reply:
But LimitGiftsRequestsAsync should care about that, doesn't it?

Reply:
You don't have that guarantee, user could set `GiftsLimiterDelay` to `0`. No limiter guarantees critical section, in fact, limiter explicitly **doesn't** guarantee that as it just load-balances enters, not synchronizes threads.